modules/core/04-channel/keeper/keeper_test.go

```diff
@@ -541,3 +542,321 @@ func (suite *KeeperTestSuite) TestUnsetParams() {
 		suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetParams(ctx)
 	})
 }
+
+func (suite *KeeperTestSuite) TestPruneAcknowledgements() {
+	var (
+		path          *ibctesting.Path
+		limit         uint64
+		upgradeFields types.UpgradeFields
+
+		// postPruneExpState is a helper function to verify the expected state after pruning. Argument expLeft
+		// denotes the expected amount of packet acks and receipts left after pruning. Argument expSequenceStart
+		// denotes the expected value of PruneSequenceStart.
+		postPruneExpState = func(expAcksLen, expReceiptsLen, expPruningSequenceStart uint64) {
+			acks := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetAllPacketAcks(suite.chainA.GetContext())
+			suite.Require().Len(acks, int(expAcksLen))
+
+			receipts := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetAllPacketReceipts(suite.chainA.GetContext())
+			suite.Require().Len(receipts, int(expReceiptsLen))
+
+			start := suite.chainA.App.GetIBCKeeper().ChannelKeeper.GetPruningSequenceStart(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+			suite.Require().Equal(start, expPruningSequenceStart)
+		}
+	)
+
+	testCases := []struct {
```

Comment:
could we add a test case where a middle sequence doesn't have an ack stored (because a timeout occurred)?